src/layouts/home-assistant.ts

```diff
@@ -182,9 +185,15 @@ export class HomeAssistantAppEl extends QuickBarMixin(HassElement) {
 
   protected hassReconnected() {
     super.hassReconnected();
+    this._checkUpdate(this.hass!.connection);
+  }
 
+  private _checkUpdate(connection: Connection) {
+    const oldVersion = this._haVersion;
+    const currentVersion = connection.haVersion;
     // If backend has been upgraded, make sure we update frontend
-    if (this.hass!.connection.haVersion !== this._haVersion) {
+    if (currentVersion !== oldVersion) {
+      this._haVersion = currentVersion;
       if (supportsServiceWorker()) {
         navigator.serviceWorker.getRegistration().then((registration) => {
```

Comment:
@bramkragten should we make _checkUpdate async to be able to wait on this check before `initializeHass`? 
In case there is an update we refresh before load anything

Reply:
The function will be sync when it doesnt support a serviceworker, and thus will reload before loading HA. If a serviceworker is supported we show a message for the user to click, and not refresh automatic.